RestCommClient/Classes/RCRegisterPushDelegate.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * TeleStax, Open Source Cloud Communications
+ * Copyright 2011-2015, Telestax Inc and individual contributors
+ * by the @authors tag.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation; either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ * For questions related to commercial use licensing, please contact sales@telestax.com.
+ *
+ */
+
+#import <Foundation/Foundation.h>
+
+@protocol RCRegisterPushDelegate<NSObject>
+@optional
+/**
+ *  @abstract Emitted when registering for push notification is successfull
+ *
+ */
+- (void)rcRegisterPushSuccessfully;
```

Comment:
Can we keep just one callback for both success & failure scenarios? It allows for less boilerplate code and also is the direction we have been taking on Android as well

Reply:
Sure. Will have one method.